analyzers/tests/SonarAnalyzer.UnitTest/Rules/ObjectCreatedDroppedTest.cs

```diff
@@ -20,17 +20,19 @@
 
 using SonarAnalyzer.Rules.CSharp;
 
-namespace SonarAnalyzer.UnitTest.Rules
+namespace SonarAnalyzer.UnitTest.Rules;
+
+[TestClass]
+public class ObjectCreatedDroppedTest
 {
-    [TestClass]
-    public class ObjectCreatedDroppedTest
-    {
-        [TestMethod]
-        public void ObjectCreatedDropped() =>
-            OldVerifier.VerifyAnalyzer(@"TestCases\ObjectCreatedDropped.cs", new ObjectCreatedDropped());
+    [TestMethod]
+    public void ObjectCreatedDropped() =>
+        new VerifierBuilder<ObjectCreatedDropped>().AddPaths("ObjectCreatedDropped.cs").Verify();
 
-        [TestMethod]
-        public void ObjectCreatedDropped_InTest() =>
-            OldVerifier.VerifyNoIssueReportedInTest(@"TestCases\ObjectCreatedDropped.cs", new ObjectCreatedDropped());
```

Comment:
Please do not drop existing tests.

There's `.AddTestReference()` and `.VerifyNoIssueReported()` to migrate it.

Reply:
> Please do not drop existing tests.
> 
> There's `.AddTestReference()` and `.VerifyNoIssueReported()` to migrate it.

I didn't thought it was useful, as I never saw it elsewhere. What is the purpose of this particular test?

Reply:
To validate the scope of the rule (main code vs. test code vs. all). Some rules do not run on test projects and vice versa. 

AFAIK, most rules do not assert that. So there probably some reason to assert that in this rule.